### PR TITLE
Add Lint check to GitHub workflows

### PR DESCRIPTION
**Pull request by hdavidethan**
New GitHub workflows are as follows:

Lint check is run whenever a push or PR request update is made.
Deploy will run only when pushes to main go through.

Lint will replace Codacy as the required status check.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: npm i
+        shell: bash
+      - name: Check TypeScript
+        run: npx tsc --noEmit
+        shell: bash
+      - name: Run ESLint
+        run: npm run lint
```
Comment (anudaweerasinghe) [inline .github/workflows/lint.yml:16]: Does this treat warnings as errors?

Comment (hdavidethan) [reply .github/workflows/lint.yml:16]: No

Comment (hdavidethan) [reply .github/workflows/lint.yml:16]: We can define which rules will cause errors in the eslintrc file

**Review by anudaweerasinghe — APPROVED** [review]
[LGTM](https://lgtm.lol/i/691)

**Pull request merged**